### PR TITLE
856 flutter screenshare for ios android

### LINKED ISSUE #856: [Flutter]Screenshare document applies for windows only. Can we have the detailed document for iOS as well?
Here the previous Flutter 5.x version screenshare. The section "Implement screen sharing on iOS" is missing in the new document. 
https://docs-preprod.agora.io/en/live-streaming-premium-legacy/screensharing_flutter?platform=Flutter

If customer follow up current document, they can't make screenshare working in Flutter iOS.

Thank you!

### PR DESCRIPTION
**Pull request by saudsami**
Updated the Flutter doc to add screen sharing on iOS, Android, and macOS.

I also updated the Windows implementation to avoid using the deprecated `startScreenCaptureByScreenRect` method.

closes #856

### CODE REVIEW

File: shared/video-sdk/develop/product-workflow/project-implementation/flutter.mdx [hunk 1]
```diff
@@ -69,44 +69,102 @@ To implement these features in your <Vpl k="CLIENT" />, take the following steps
...
+            audioParams: ScreenAudioParameters(
+                sampleRate: 16000, channels: 2, captureSignalVolume: 100),
+            captureVideo: true,
+            videoParams: ScreenVideoParameters(
+                dimensions: VideoDimensions(height: 1280, width: 720),
+                frameRate: 15,
+                bitrate: 600)));  
+        ```
+        
+        :warning: **Note**:
```
Comment (billy-the-fish) [inline shared/video-sdk/develop/product-workflow/project-implementation/flutter.mdx:155]: The iOS stuff should be part of the prerequisites. Can you please update the iOS prereqs for flutter to include the platform version. 


And the Android bit as a step in the projet setup please.

**Review by billy-the-fish — CHANGES_REQUESTED** [review]
Nearly there.

File: shared/video-sdk/develop/product-workflow/project-implementation/flutter.mdx (hunk 1 above)
Comment (saudsami) [reply shared/video-sdk/develop/product-workflow/project-implementation/flutter.mdx]: Done

**Review by billy-the-fish — APPROVED** [review]
LGTM.

**Pull request merged**